reframe/frontend/cli.py

```diff
@@ -229,6 +238,18 @@ def main():
     # Init modules system
     init_modules_system(system.modules_system)
 
+    try:
+        if options.module_map_file:
+            get_modules_system().load_mapping_from_file(
+                options.module_map_file)
+
+        if options.module_mappings:
+            for m in options.module_mappings:
+                get_modules_system().load_mapping(m)
+
+    except (ReframeError, OSError) as e:
+        raise ReframeError('could not set module mapping: %s' % e)
```

Comment:
You should not raise an exception here. You are just handling them here and present the error to the user. Have you tried the output by passing an invalid mapping. It's like crashing. You should do instead:

```python
    except (ReframeError, OSError) as e:
        printer.error('could not load module mappings: %s' % e)
        sys.exit(1)
```